paper-server/patches/sources/net/minecraft/world/entity/projectile/FishingHook.java.patch

```diff
@@ -42,6 +42,51 @@
                      return;
                  }
              } else {
+@@ -169,15 +_,23 @@
+             boolean flag = f > 0.0F;
+             if (this.currentState == FishingHook.FishHookState.FLYING) {
+                 if (this.hookedIn != null) {
+-                    this.setDeltaMovement(Vec3.ZERO);
+-                    this.currentState = FishingHook.FishHookState.HOOKED_IN_ENTITY;
+-                    return;
++                    // Paper start - Add FishHookStateChangeEvent. #HOOKED_ENTITY
++                    if (new io.papermc.paper.event.entity.FishHookStateChangeEvent((org.bukkit.entity.FishHook) getBukkitEntity(), org.bukkit.entity.FishHook.HookState.HOOKED_ENTITY).callEvent()) {
++                        this.setDeltaMovement(Vec3.ZERO); // before this.setDeltaMovement(Vec3.ZERO);
++                        this.currentState = FishingHook.FishHookState.HOOKED_IN_ENTITY;
++                        return;
++                    }
```

Comment:
Avoid changing the indentation of these lines to help with diff. 
So you would have a paper comment on the if statement, then the bracket 
So
if (...) { // Paper - blah
...
...
..
} // Paper - blah

This way you are not changing the code inside of the if, and it wraps the diff cleanly.

Reply:
Thanks for the review! That makes a lot of sense!